Code/Framework/AzCore/AzCore/Serialization/SerializeContext.cpp

```diff
@@ -888,2398 +878,2612 @@ namespace AZ
-            return -1;
+            return this;
         }
-    #endif // AZ_ENABLE_TRACING
-
-        DataElementNode node;
-        node.m_element.m_name = name;
-        node.m_element.m_nameCrc = nameCrc;
-        node.m_element.m_id = classData.m_typeId;
-        node.m_classData = &classData;
-        node.m_element.m_version = classData.m_version;
-
-        m_subElements.push_back(node);
-        return static_cast<int>(m_subElements.size() - 1);
+        m_classData->second.m_container = dataContainer;
+        return this;
     }
 
-    int SerializeContext::DataElementNode::AddElement(SerializeContext& sc, AZStd::string_view name, GenericClassInfo* genericClassInfo)
+    //=========================================================================
+    // ClassBuilder::PersistentId
+    //=========================================================================
+    SerializeContext::ClassBuilder* SerializeContext::ClassBuilder::PersistentId(ClassPersistentId persistentId)
     {
-        (void)sc;
-        AZ_Assert(!name.empty(), "Empty name is an INVALID element name!");
-
-        if (!genericClassInfo)
+        if (m_context->IsRemovingReflection())
         {
-            AZ_Assert(false, "Supplied GenericClassInfo is nullptr. ClassData cannot be retrieved.");
-            return -1;
+            return this; // we have already removed the class data.
         }
-
-        u32 nameCrc = Crc32(name.data());
-        DataElementNode node;
-        node.m_element.m_name = name.data();
-        node.m_element.m_nameCrc = nameCrc;
-        node.m_element.m_id = genericClassInfo->GetClassData()->m_typeId;
-        node.m_classData = genericClassInfo->GetClassData();
-        node.m_element.m_version = node.m_classData->m_version;
-
-        m_subElements.push_back(node);
-        return static_cast<int>(m_subElements.size() - 1);
+        m_classData->second.m_persistentId = persistentId;
+        return this;
     }
 
     //=========================================================================
-    // ReplaceElement
+    // ClassBuilder::SerializerDoSave
     //=========================================================================
-    int SerializeContext::DataElementNode::ReplaceElement(SerializeContext& sc, int index, const char* name, const Uuid& id)
+    SerializeContext::ClassBuilder* SerializeContext::ClassBuilder::SerializerDoSave(ClassDoSave doSave)
     {
-        DataElementNode& node = m_subElements[index];
-        if (node.Convert(sc, name, id))
-        {
-            return index;
-        }
-        else
+        if (m_context->IsRemovingReflection())
         {
-            return -1;
+            return this; // we have already removed the class data.
         }
+        m_classData->second.m_doSave = doSave;
+        return this;
     }
 
     //=========================================================================
-    // SetName
-    // [1/16/2013]
+    // EnumerateInstanceConst
+    // [10/31/2012]
```

Comment:
That is so long ago

Reply:
Those comments are also useless.
The data provides the reader with no meaningful information.

The name of the function is extra point failure for mismatched documentation as it would need to be kept in sync as well, for no meaningful information.
This makes the comments have an overall net negative value to the code.